Doc/library/annotationlib.rst

```diff
@@ -421,6 +427,94 @@ Functions
 
    .. versionadded:: 3.14
 
+.. _functools-objects-annotations:
```

Comment:
I feel like this doesn't belong in the annotationlib docs, maybe in the functools ones.

Reply:
I have a different idea here. plz correct me if I'm wrong

It seems that we recommend people use `annotationlib` to process the annotation here. I prefer add a more detail document in here and reference this in functools

Reply:
annotationlib is a general tool, functools objects aren't special for it. annotationlib should document how unrelated objects (such as partial) can support annotations, but it shouldn't have documentation specific to those objects.